scripts-available/CDB_PG_Federated_Tables.sql

```diff
@@ -0,0 +1,295 @@
+----------------------------------------------------------------------
+-- Federated Tables management functions
+----------------------------------------------------------------------
+
+-- Take a config jsonb and transform it to an input suitable for
+-- _CDB_SetUp_User_PG_FDW_Server
+CREATE OR REPLACE FUNCTION @extschema@.__ft_credentials_to_user_mapping(input_config jsonb)
+RETURNS jsonb
+AS $$
+DECLARE
+    user_mapping jsonb;
+BEGIN
+    user_mapping := json_build_object('user_mapping',
+        jsonb_build_object(
+            'user', input_config->'credentials'->'username',
+            'password', input_config->'credentials'->'password'
+        )
+    );
+    RETURN (input_config - 'credentials')::jsonb || user_mapping;
+END
+$$
+LANGUAGE PLPGSQL IMMUTABLE PARALLEL SAFE;
+
+
+-- Take a config jsonb as input and return it augmented with default
+-- options
+CREATE OR REPLACE FUNCTION @extschema@.__ft_add_default_options(input_config jsonb)
+RETURNS jsonb
+AS $$
+DECLARE
+    default_options jsonb := '{
+        "extensions": "postgis",
```

Comment:
Do we want to have `postgis` on by default without checking the remote server? Should we leave it out until we can check it in a following iteration?

Reply:
I'll leave it "as is" as our initial target is still databases with postgis installed.